src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -163,17 +176,14 @@ public void actionPerformed(ActionEvent e) {
         final String s = (String) JOptionPane.showInputDialog(
             UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
             null, null, "");
-        if (s != null && s.length() > 0) {
-          //if alphanumeric
-          if(s.matches("[a-zA-Z0-9]+")) {
-            clientContext.user.addUser(s);
-            UserPanel.this.getAllUsers(listModel);
-          } else {
-            JOptionPane.showMessageDialog(UserPanel.this,
-                "User not created. Alphanumeric characters only, with no spaces. Please try again.",
-                "User Not Created",
-                JOptionPane.ERROR_MESSAGE);
-          }
+        if (validator.isValidUserName(s)) {
```

Comment:
I mean we are talking about constants so small here it really doesn't matter. This program has other features that are performing in worse than linear time, this function will make absolutely 0 difference in the runtime of the program. It's more about functionality, clarity, and maintainability